lib/ts/I_Layout.h

```diff
@@ -31,80 +31,93 @@
 
   /**
    update the sysconfdir to a test conf dir
    */
-  void update_sysconfdir(const char *dir);
+  void update_sysconfdir(ts::string_view dir);
+
+  /**
+   Return file path relative to dir
+   Function for when the pass in parameter dir is type ats_scoped_str
+   Example usage: Layout::relative_to(default_layout()->sysconfdir, "foo.bar");
+
+  */
+  static char *relative_to(ts::string_view dir, ts::string_view file);
+
+  static std::string relative_to_overload(ts::string_view dir, ts::string_view file);
 
   /**
    Return file path relative to Layout->prefix
    Store the path to buf. The buf should be large eough to store
    PATH_NAME_MAX characters
 
    */
-  void relative(char *buf, size_t bufsz, const char *file);
+  void relative(char *buf, size_t bufsz, ts::string_view file);
 
   /**
    Return file path relative to dir
-   Memory is allocated, so use ats_free() when no longer needed
+   Store the path to buf. The buf should be large eough to store
+   PATH_NAME_MAX characters
    Example usage: Layout::relative_to(default_layout()->sysconfdir, "foo.bar");
 
   */
-  static char *relative_to(const char *dir, const char *file);
+  static void relative_to(char *buf, size_t bufsz, ts::string_view dir, ts::string_view file);
 
   /**
-   Return file path relative to dir
-   Store the path to buf. The buf should be large eough to store
-   PATH_NAME_MAX characters
-   Example usage: Layout::relative_to(default_layout()->sysconfdir, "foo.bar");
-
+   Check for TS_RUNROOT and yaml file
+   return the yaml file path if correct and found
   */
-  static void relative_to(char *buf, size_t bufsz, const char *dir, const char *file);
+  static std::string check_runroot();
 
   /**
    Creates a Layout Object with the given prefix.  If no
    prefix is given, the prefix defaults to the one specified
    at the compile time.
 
   */
-  static void create(const char *prefix = 0);
+  static void create(ts::string_view prefix = "");
 
   /**
    Returns the Layout object created by create_default_layout().
 
   */
   static Layout *get();
 
-  char *prefix        = nullptr;
-  char *exec_prefix   = nullptr;
-  char *bindir        = nullptr;
-  char *sbindir       = nullptr;
-  char *sysconfdir    = nullptr;
-  char *datadir       = nullptr;
-  char *includedir    = nullptr;
-  char *libdir        = nullptr;
-  char *libexecdir    = nullptr;
-  char *localstatedir = nullptr;
-  char *runtimedir    = nullptr;
-  char *logdir        = nullptr;
-  char *mandir        = nullptr;
-  char *infodir       = nullptr;
-  char *cachedir      = nullptr;
+  std::string prefix;
+  std::string exec_prefix;
+  std::string bindir;
+  std::string sbindir;
+  std::string sysconfdir;
+  std::string datadir;
+  std::string includedir;
+  std::string libdir;
+  std::string libexecdir;
+  std::string localstatedir;
+  std::string runtimedir;
+  std::string logdir;
+  std::string mandir;
+  std::string infodir;
+  std::string cachedir;
```

Comment:
maybe.. but it does not buy us anything make the members const. The assessors deal with or should as these items should not be public.